test/models/test_mamba.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from tinygrad.helpers import CI
+from tinygrad import Device
+from examples.mamba import Mamba, generate
+from transformers import AutoTokenizer
+
+PROMPT = 'Why is gravity '
+TOKENIZER = AutoTokenizer.from_pretrained("EleutherAI/gpt-neox-20b")
+
+@unittest.skipIf(CI and Device.DEFAULT in ["CLANG"], "Inference hack broken on CLANG")
+class TestMamba(unittest.TestCase):
+  def test_mamba_130M(self):
+    OUT_130M = '''Why is gravity \nnot a good idea?\n\nA:'''
+    model = Mamba.from_pretrained('130m')
+    tinyoutput = generate(model, TOKENIZER, PROMPT, n_tokens_to_gen=10)
+    self.assertEqual(OUT_130M, tinyoutput)
+    del model
+  def test_mamba_370M(self):
+    OUT_370M = '''Why is gravity \nso important?\nBecause it's the only'''
```

Comment:
How did you verify this is correct?

Reply:
Ran it on a troch implementation: https://github.com/reddyn12/tinygrad/blob/e9dfa6fe57e689df4e5d342989cd236bf317f407/examples/compare/mamba/mambaTorch.py

Reply:
Why is it putting in the \n? I have never seen an LLM do that

Reply:
How else can a new line happen?

Reply:
I can try with pure "enter" but shouldn't be a difference right?

Reply:
That's not what I mean. Why is `\n` the next symbol?

Reply:
Oh, maybe a training thing? poem roleplay?

Reply:
I simply don't believe this. Does mamba on huggingface do this?

Reply:
Ya:
```python 
from transformers import AutoModelForCausalLM , AutoTokenizer

model = AutoModelForCausalLM.from_pretrained('Q-bert/Mamba-130M', trust_remote_code=True)
tokenizer = AutoTokenizer.from_pretrained('Q-bert/Mamba-130M')

text = 'Why is gravity '

input_ids = tokenizer.encode(text, return_tensors="pt")

output = model.generate(input_ids, max_length=20, num_beams=1, no_repeat_ngram_size=2)

generated_text = tokenizer.decode(output[0], skip_special_tokens=True)

print(generated_text)
```

Reply:
If I take the trailing space off of the prompt, it does it in one line.